include/nbl/video/CVulkanImage.h

```diff
@@ -36,6 +36,24 @@ class CVulkanImage : public IGPUImage
 
 		inline size_t getBoundMemoryOffset() const override { return m_memBindingOffset; }
 
+		void* getExternalHandle() override
+		{
+			if (m_cachedExternalHandle)
+				return m_cachedExternalHandle;
+
+			auto& ccp = getCachedCreationParams();
+
+			if (ccp.externalHandleTypes.value)
+			{
+				if (ccp.externalHandle)
+					return m_cachedExternalHandle = ccp.externalHandle;
+
+				return m_cachedExternalHandle = getOriginDevice()->getExternalHandle(this);
+			}
```

Comment:
I think you can more this to `IDeviceMemoryBacked` and query it at `bind()` time